### ISSUE TITLE
Using AOE gold data -  "Failed to preload DRS archives: IO Error"

### ISSUE THREAD

**Issue by EggsSeriously** [angered-ghandi/OpenAOE#1]
Using data from the AOE gold disk, which included the Rise of rome exspansion.

After compiling the source and placing relevent files, running cargo run nets the following response from terminal : 

  Running `target/debug/open_aoe`
  Loading "data/border.drs"...
  Failed to preload DRS archives: IO Error
  error: Process didn't exit successfully: `target/debug/open_aoe` (exit code: 1)

This error applies weather I place the data from AOE in openaoe/game/data , openaoe/data, or at the root of the folder.

**Comment by angered-ghandi** [comment]
Thanks for trying it out!

I think something must be different between our setups. I was able to load it up with the Gold Edition data on my end. Here's what mine looks like:

```
$ ls
Cargo.lock  Cargo.toml  crates  data  LICENSE  README.md  src  target  tools

$ ls data
border.drs  empires.dat  graphics.drs  interfac.drs  sounds.drs  terrain.drs  test2.scn  tileedge.dat

$ cargo run
     Running `target/debug/open_aoe`
Loading "data/border.drs"...
Loading "data/graphics.drs"...
Loading "data/interfac.drs"...
Loading "data/sounds.drs"...
Loading "data/terrain.drs"...
Loaded empires.dat
Loaded test2.scn
```

Things to check:
- Did you copy the files from data/ or data2/? I think the contents of data/ are the original AOE while data2/ has the additional ROR content. I haven't tested against any of the ROR files yet (start simple, right?)
- If you're on a case-sensitive filesystem, are the data file names all lowercase?
- Do you have permissions to read the file?

Note, you'll need to create a scenario in the original game and copy it into data/test2.scn or it'll fail on that last step. I haven't committed any test data.

Pointing it at the game data is a little weird at present. I'm thinking there should be some command-line options to point it to the game data location on the original CD or something.

Looks like I should tack additional information on to the IO error message :)

**Comment by EggsSeriously** [comment]
I'm more than glad to try this out, this project is cool!  Now, for the thinges I checked : 
- Yes, my file system is case sensitive.  I used "`ls | while read upName; do loName=`echo "${upName}" | tr '[:upper:]' '[:lower:]'`; mv "$upName" "$loName"; done`" to rename everything to lowercase. I tried to copy your DIR file for file, rather than the entire DATA folder. I also did use DATA not data2. Here is my current /data dir : 

> $ ls
> border.drs  empires.dat  graphics.drs  interfac.drs  linuxtest.scn  sounds.drs  terrain.drs  tileedge.dat

and here is the output I get now : http://pastebin.com/d37WmBiy

EDIT: Oh gosh. I'm not a smart cookie. You said to create a scenario and name is test2.scn. In my infinite wisdom, I named it linux test.scn. Sorries! I got it work, and it displayed the north edge of the scenario I made. Is it possible to look around the map? 

I guess I'll start learning rust now.

TL:DR : You were right. The files needed to be lower case. That fixed everything.

Not sure how to mark this as solved..

**Comment by angered-ghandi** [comment]
I think there are a few issues we can pull out of this:
- Since the game files tend to be in ALLCAPS, the game should probably look for both variants
- The IO Error is total garbage right now--need to investigate how to make that easier to see what's going on
- The test2.scn thing is temporary while I'm implementing/testing terrain blending. I should probably allow for the name of the scenario to be specified on the command line.

I haven't implemented any kind of camera movement yet--but there are some hardcoded numbers in the main.rs that represent the camera offset. If you wanted to move the camera, you'd have to modify and recompile. I just haven't implemented input of any kind yet, so while the camera support exists, there's no way for the game to know the user is requesting anything yet.

I'll file these issues and close this. Thanks!

**Comment by angered-ghandi** [comment]
Issues filed.
